packages/rich-text/src/plugins/Text/__tests__/createTextPlugin.test.tsx

```diff
@@ -0,0 +1,294 @@
+      input: (
+        <fragment>
+          <hp>
+            <cursor />
+          </hp>
+          <htable>
+            <htr>
+              <htd>
+                <hp>p1</hp>
+              </htd>
+            </htr>
+          </htable>
+        </fragment>
+      ),
+      expected: (
+        <htable>
+          <htr>
+            <htd>
+              <hp>
+                <cursor />
+                p1
+              </hp>
+            </htd>
+          </htr>
+        </htable>
+      ),
+    },
+  ];
+
+  const render = (children: any) => (
+    <editor>
+      {children}
+      <hp>
+        <htext />
+      </hp>
+    </editor>
+  );
+
+  for (const t of tests) {
+    // eslint-disable-next-line jest/valid-title
+    test(t.title, () => {
+      const { editor } = createTestEditor({
+        input: render(t.input),
+      });
+
+      // Equivalent of pressing backspace
+      editor.deleteBackward('character');
+
+      assertOutput({
+        editor,
+        expected: render(t.expected),
+      });
+    });
+  }
+});
+
+describe('delete forward', () => {
+  const tests = [
+    {
+      title: 'deletes a character of the text inside li',
+      input: (
+        <hul>
+          <hli>
+            <hp>
+              <cursor />
+              p1
+            </hp>
+          </hli>
+        </hul>
+      ),
+      expected: (
+        <hul>
+          <hli>
+            <hp>
+              <cursor />1
+            </hp>
+          </hli>
+        </hul>
+      ),
+    },
+    {
+      title:
+        'deletes the empty paragraph at the beginning of the RTE followed by another paragraph',
+      input: (
+        <fragment>
+          <hp>
+            <cursor />
+          </hp>
+          <hp>text</hp>
+        </fragment>
+      ),
+      expected: (
+        <hp>
+          <cursor />
+          text
+        </hp>
+      ),
+    },
+    {
+      title: 'deletes the empty paragraph at the beginning of the RTE followed by li',
+      input: (
+        <fragment>
+          <hp>
+            <cursor />
+          </hp>
+          <hul>
+            <hli>
+              <hp>p1</hp>
+            </hli>
+          </hul>
+        </fragment>
+      ),
+      expected: (
+        <hul>
+          <hli>
+            <hp>
+              <cursor />
+              p1
+            </hp>
+          </hli>
+        </hul>
+      ),
+    },
+    {
+      title: 'deletes the empty paragraph at the beginning of the RTE followed by a blockquote',
+      input: (
+        <fragment>
+          <hp>
+            <cursor />
+          </hp>
+          <hquote>
+            <hp>p1</hp>
+          </hquote>
+        </fragment>
+      ),
+      expected: (
+        <hquote>
+          <hp>
+            <cursor />
+            p1
+          </hp>
+        </hquote>
+      ),
+    },
+    {
+      title: 'deletes the empty paragraph at the beginning of the RTE followed by a table',
```

Comment:
This case doesn't actually work when you try it yourself 🤔

Reply:
I tested it locally now, it works manually.

Reply:
Try it [here](https://deploy-preview-1002--contentful-field-editors.netlify.app/rich-text) (the preview link for this PR). Pressing backspace works but delete doesn't

Reply:
seen with the team in PH, they approved it works and that deleting the empty paragraph should be working.

Reply:
I think what @z0al meant is that pressing the key `delete` doesn't delete the empty paragraph. Only the `backspace` key. You can simulate the `delete` key by pressing `fn + backspace`.

Reply:
Should we support delete key as well?

Reply:
Maybe Slate is not triggering when pressing delete in this case?

Reply:
Cezar is right. I meant the `delete` key, not backspace. 

Tbh, I'm more concerned about the false positive result of the test. We don't want a test that doesn't cover the real usage as it will be just misleading and bugs will slip through. Are we sure Slate always calls `deleteForward(character)` when pressing the `DELETE` key?

Reply:
I see, I understand the case now. I just tested whether that function is being called with a simple console.log in that specific case and it's actually not, which explains why it's not working in the manual test. I need to investigate more to understand which function is actually overriding that one in this case.